File: linear_search/java/search_2d_max.java
package linear_search.java;

public class search_2d_max {
    public static void main(String[] args){
        int[][] array = {
            {10,21,22},
            {22,100},
            {77,200,80}
        };

        int ans = search_2d_max(array);
        System.out.println(ans);

}


static int search_2d_max(int[][] array){
    int max_value = array[0][0];
    for(int row = 0; row<array.length; row ++){
        for(int column = 0; column<array[row].length; column++){
            if (max_value < array[row][column]){
                max_value = array[row][column];
            }
        }
    }
    return max_value;
}
}

File: linear_search/python/search_2d_max.py
def search_2d_max(array):
    max_value = array[0][0]
    for i in range(len(array)):
        for j in range(len(array[i])):
            if array[i][j] > max_value:
                max_value = array[i][j]

    return max_value


array = [
    [10,24,54],
    [11,78,72],
    [5,8,45]
]

print(search_2d_max(array))
